table/tables/partition_test.go

```diff
@@ -714,6 +2170,87 @@ func TestIssue31721(t *testing.T) {
 	tk.MustExec("select * from t_31721 partition(p0, p1) where col1 != 2;")
 }
 
+func TestKeyPartitionTableDDL(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("create database partitiondb3")
+	defer tk.MustExec("drop database partitiondb3")
+	tk.MustExec("use partitiondb3")
+	tk.MustExec("CREATE TABLE tkey14 (\n" +
+		"col1 INT NOT NULL," +
+		"col2 INT NOT NULL," +
+		"col3 INT NOT NULL," +
+		"col4 INT NOT NULL," +
+		"primary KEY (col1,col3)\n" +
+		")" +
+		"PARTITION BY HASH(col3) PARTITIONS 4")
```

Comment:
Should it be HASH or KEY partitioned tables in this test?

Reply:
good catch.